src/screens/Instructor/InstPlaylist/components/PlaylistInfo/index.js

```diff
@@ -40,23 +40,44 @@ function PlaylistInfoWithRedux(props) {
 
   const [editing, setEditing] = useState(false);
   const [inputValue, setInputValue] = useState(playlist.name);
-
+  const [phraseList, setPhraseList] = useState(playlist.jsonMetadata.phraseList || []);
+  
   const handleEdit = () => setEditing(true);
 
   const onInputChange = ({ target: { value } }) => {
     setInputValue(value);
   };
 
-  const handleCancelRename = () => setEditing(false);
-
-  const handleRename = async () => {
+  const handleCancelEdit = () => setEditing(false);
+
+  // const handleRename = async () => {
+  //   if (!inputValue) return;
+  //   const playlist_ = await Playlist.rename(playlist, inputValue);
+  //   if (playlist_) {
+  //     dispatch({type: 'instplaylist/setPlaylist', payload: playlist_.toObject()});
+  //   }
+  //   handleCancelEdit();
+  // };
+  
+  const handleSaveEdits = async () => {
     if (!inputValue) return;
-    const playlist_ = await Playlist.rename(playlist, inputValue);
+    const playlist_ = await Playlist.edit(playlist, inputValue, phraseList);
```

Comment:
Should be in setPlaylist effect function

Reply:
Same for this one. I should create new effect functions for all places where api calls have been made outside of the model and refactor the code for InstPlaylist screen after this